### ISSUE TITLE
make oracledb_session platform independent

### ISSUE THREAD

**Issue by ehaselwanter** [inspec/inspec#3255]
### 🎛 Description

🐛 bug report

The documentation states the oracledb_session is platform independent, but it fails to run on windows.

### 🌍 InSpec and Platform Version

macOS

```
chef -v
Chef Development Kit Version: 3.1.0
chef-client version: 14.2.0
delivery version: master (6862f27aba89109a9630f0b6c6798efec56b4efe)
berks version: 7.0.4
kitchen version: 1.22.0
inspec version: 2.1.72
```

but with latest changes from https://github.com/inspec/inspec/commit/a16877f4273c2952270d698f7f8ae3cc5ff9915b

### 🤔 Replication Case

```
sql = oracledb_session(user: 'u', password: 'p', service: 'XE', sqlplus_bin: 'sqlplus', as_os_user: nil)

describe sql.query("select * from user_tablespaces where tablespace_name LIKE 'TEST'").row(0).column('tablespace_name') do
  its('value') { should eq 'TEST' }
end
```

### 💁 Possible Solutions

basically add powershell heredoc to https://github.com/inspec/inspec/blob/master/lib/resources/oracledb_session.rb#L69

```
      # use heredoc to ship the sql as a multiline string
      # we need to do it differently in powershell (prefix) and bash (postfix)
      if inspec.os.windows?
        sql_prefix = "@'\n#{opts}\n#{query}\nexit\n'@ | "
        sql_postfix = ''
      else
        sql_postfix = " <<EOC\n#{opts}\n#{query}\nEXIT\nEOC"
        sql_prefix = ''
      end

      if @db_role.nil? # rubocop:disable Style/ConditionalAssignment
        command = %(#{sql_prefix}#{bin} #{@user}/#{@password}@#{@host}:#{@port}/#{@service}#{sql_postfix})
      elsif @su_user.nil?
        command = %(#{sql_prefix}#{bin} "#{@user}"/"#{@password}"@#{@host}:#{@port}/#{@service} as #{@db_role}#{sql_postfix})
      else
        command = %(su - #{@su_user} -c "env ORACLE_SID=#{@service} #{bin} / as #{@db_role}#{sql_postfix})
      end
```

### 💻 Stacktrace

none at hand

**Comment by bytemarcz** [comment]
Hello, @ehaselwanter did you get this resource to work on a unix/linux system?

**Comment by ehaselwanter** [comment]
@maczamora yes. you need to decide what you use as the executable. We decided to go with sqlplus. and the other thing which might hit you is that you need to fix the LD_LIBRARY path for sqlplus, since you are coming in over ssh and it might be missing.

something like:

```ruby
    build_essential 'oci8 bin install'

    package %w(libaio oracle-instantclient12.2-sqlplus oracle-instantclient12.2-devel)

    oracle_dynamic_lib_path = '/usr/lib/oracle/12.2/client64/lib/'

    file '/etc/ld.so.conf.d/oracle_instaclient.conf' do
      content oracle_dynamic_lib_path
    end

    execute 'Configure dynamic linker run-time bindings' do
      command '/sbin/ldconfig'
      not_if "/sbin/ldconfig -p |grep #{oracle_dynamic_lib_path}"
    end

    link '/usr/bin/sqlplus' do
      to '/usr/bin/sqlplus64'
      only_if 'test -f /usr/bin/sqlplus64'
    end
```

**Comment by bytemarcz** [comment]
@ehaselwanter so if the executable for sqlplus and LD_Library aren't a factor would you recommend to try running the test from a unix/linux based system? I'm currently running from host (Win) sshing to AIX IBM server where the db is located.

**Comment by kekaichinose** [comment]
Fixed as part of #4654. Closing.
